src/priceClient/adapters/index.ts

```diff
@@ -0,0 +1,2 @@
+export * as across from "./across";
```

Comment:
Minor note for the future (nothing to change here!): I know the rest of the sdk works this way, but namespacing all the components like this may have a negative impact on tree-shaking, as I'm not sure webpack can infer which parts of the application are used if they are all exported under namespaces.

In general, we may want to reconsider the approach unless we are confident it won't stop webpack from dropping unused symbols.

Reply:
Interesting - had no idea about this. Is it the `export *` or `as across` that causes issues?

I _think_ we can actually get away without the `export *` statement, since we could just mark the PriceFeed class as the default export. It's still nice to have that exported under the `across` namespace though, because the `PriceFeed` name is reused in the coingecko class (i.e. `across.PriceFeed`, `coingecko.PriceFeed`).

Reply:
@mrice32 To pick up on this one, do we want to note this for future consideration in a Linear issue?

Reply:
@pxrl I'm not 100% sure about how tree-shaking works.

My basic understanding is that importing like this allows webpack to shake all parts of the imported code that isn't MyClass:
```ts
import { MyClass } from "@myscope/mypackage";

...

new MyClass();
```

Alternatively, if MyClass, is wrapped up in a namespace with 10 other classes, you would be forced to do this:

```ts
import { mynamespace } from "@myscope/mypackage";

...

new mynamespace.MyClass();
```

I think that would mean that all of mynamespace would be bundled because webpack can't (or just isn't sophisticated enough) to know which part you're going to use.

This is probably something we should test before making any changes, but I know that tree-shaking is very sensitive to things like this.